diff -Naur orig/field_formatter_settings.module new/field_formatter_settings.module
--- orig/field_formatter_settings.module	2023-12-05 23:50:17.053164680 +0530
+++ new/field_formatter_settings.module	2023-12-05 23:49:27.370536815 +0530
@@ -31,7 +31,7 @@
 
   // Fetch the instance settings and default display settings.
   $info = field_info_instance($entity_type, $field_name, $bundle);
-  $defaults = $info['display']['default']['settings'];
+  $defaults = $info['display']['default']['settings'] ?? [];
 
   if (isset($info['display'][$view_mode]['settings'])) {
     return $info['display'][$view_mode]['settings'] + $defaults;
@@ -79,7 +79,7 @@
 
     $instance['display'][$view_mode]['type'] = $formatter_type;
     $formatter = field_info_formatter_types($formatter_type);
-    $instance['display'][$view_mode]['module'] = $formatter['module'];
+    $instance['display'][$view_mode]['module'] = isset($formatter['module']) ? $formatter['module'] : '';
     $instance['display'][$view_mode]['settings'] = $settings;
 
     // Base button element for the various formatter settings actions.
@@ -142,7 +142,7 @@
     }
     else {
       // Display a summary of the current formatter settings.
-      $summary = module_invoke($formatter['module'], 'field_formatter_settings_summary', $field, $instance, $view_mode);
+      $summary = isset($formatter['module']) ? module_invoke($formatter['module'], 'field_formatter_settings_summary', $field, $instance, $view_mode) : '';
 
       // Allow other modules to alter the formatter summary.
       $context = array(
